### ISSUE TITLE
CheckButton size problem

### ISSUE THREAD

**Issue by fatfatson** [cocos2d/cocos2d-x#13524]
the frontCross part will scale itself to the "_contentSize" when show 
( see in: 
void AbstractCheckButton::frontCrossTextureScaleChangedWithSize()
)
and change to "_backgroundTextureScaleX/Y" when hide
( see in:
void AbstractCheckButton::onPressStateChangedToNormal()
)

it will cause problem if we use a texutre smaller than the background part:
1. the frontCross texture is scaled without real need.
2. the frontCross will show as a bigger size at first ( scaled to content size ), but if we click to unselect it and then select it again by "setSelected" call instead of actually clicking, it will show as a smaller(normal) size finally. because when deselect by click, it has scaled to  "_backgroundTextureScaleX/Y"(almost 1.0f),  but "setSelected" call won't active the SizeChange notify for recovery.

adivse:
why use two parts for a checkbox (front and background) ? one is enough, we can make a background image with a cross and another one without, just as normal button. the frontCross part is so small and has litte value for reuse, but increase the complexity.

**Comment by pandamicro** [comment]
@neokim @zilongshanren thoughts ?

**Comment by neokim** [comment]
I agree with @fatfatson that the frontCross part increases the complexity.
Actually it is not a check **button**. It looks that it is designed only for a check **box** even though its logic could be used for any check button. In current implementation it cannot be used for following case of check button.
![check_button](https://cloud.githubusercontent.com/assets/3601028/9594401/67c5c252-5095-11e5-8d2f-c5e4879f6af8.png)
To make it work with `CheckBox`, I have had to write a code to remove the background image when it is checked.

My suggestion is to modify `CheckBox` (actually `AbstractCheckButton`) to subclass `Button`. Because the logics for image changing by its state is already in there and `CheckBox` can only override some state related logics and event dispatch. And change the name to `CheckButton`.

If it is not possible because it breaks existing APIs, how about writing a new class `CheckButton`? I can help writing it.
